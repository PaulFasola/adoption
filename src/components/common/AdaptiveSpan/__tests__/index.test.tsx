import React from 'react';
import toJson from "enzyme-to-json";
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { configure, shallow } from 'enzyme';
import { AdaptiveSpan } from '..';

configure({ adapter: new Adapter() });

describe('AdaptiveSpan Component', () => {
  it('should renders', () => {
    const wrapper = shallow(
      <AdaptiveSpan
        text="Foobar"
      />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should have a title when showTitle is true', () => {
    const wrapper = shallow(
      <AdaptiveSpan
        showTitle
        text="Foobar"
      />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should still render if a mapped value is forced to undefined', () => {
    const wrapper = shallow(
      <AdaptiveSpan
        showTitle
        text="<u>Foobar is {foobar} and void is {void}</u>"
        mapping={[
          { tag: '{foobar}', value: 'foobar' },
          { tag: '{void}', value: undefined! }
        ]}
      />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should apply general styling correctly', () => {
    const boldSpan = shallow(
      <AdaptiveSpan
        showTitle
        text="Foobar"
        style="bold" />);

    expect(toJson(boldSpan)).toMatchSnapshot();

    const wrapper = shallow(
      <AdaptiveSpan
        showTitle
        text="Foobar"
        style="italic" />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should replace a tag accordingly', () => {
    const wrapper = shallow(
      <AdaptiveSpan
        showTitle
        text="Foo{test}"
        mapping={[
          { tag: '{test}', value: 'bar' }
        ]} />);

    expect(wrapper.html()).toEqual('<span title=\"Foobar"\>Foobar</span>');
  });

  it('should replace occurencies matching tags defined in mapping property correctly', () => {
    const wrapper = shallow(
      <AdaptiveSpan
        showTitle
        text="{test1} is not {test2}, but {test1} is {test1}"
        mapping={[
          { tag: '{test1}', value: 'foo' },
          { tag: '{test2}', value: 'bar' }
        ]}
      />);

    expect(wrapper.html()).toEqual('<span title=\"foo is not bar, but foo is foo\">foo is not bar, but foo is foo</span>');
  });

  it('should replace occurencies and apply the requested styles', () => {
    const wrapper = shallow(<AdaptiveSpan
      showTitle
      text="{test1} is not {test2}, and {test2} is not {test3}."
      mapping={[
        { tag: '{test1}', value: 'foo', style: 'bold' },
        { tag: '{test2}', value: 'bar', style: 'italic' },
        { tag: '{test3}', value: 'foobar' }
      ]} />);

    expect(wrapper.html()).toEqual('<span title=\"foo is not bar, and bar is not foobar.\"><b>foo</b> is not <i>bar</i>, and <i>bar</i> is not foobar.</span>');
  });

});