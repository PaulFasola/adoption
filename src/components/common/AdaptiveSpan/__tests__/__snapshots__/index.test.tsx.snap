// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdaptiveSpan Component should apply general styling correctly 1`] = `
<span
  dangerouslySetInnerHTML={
    Object {
      "__html": "<b>Foobar</b>",
    }
  }
  title="Foobar"
/>
`;

exports[`AdaptiveSpan Component should apply general styling correctly 2`] = `
<span
  dangerouslySetInnerHTML={
    Object {
      "__html": "<i>Foobar</i>",
    }
  }
  title="Foobar"
/>
`;

exports[`AdaptiveSpan Component should have a title when showTitle is true 1`] = `
<span
  dangerouslySetInnerHTML={
    Object {
      "__html": "Foobar",
    }
  }
  title="Foobar"
/>
`;

exports[`AdaptiveSpan Component should renders 1`] = `
<span
  dangerouslySetInnerHTML={
    Object {
      "__html": "Foobar",
    }
  }
/>
`;

exports[`AdaptiveSpan Component should still render if a mapped value is forced to undefined 1`] = `
<span
  dangerouslySetInnerHTML={
    Object {
      "__html": "<u>Foobar is foobar and void is </u>",
    }
  }
  title="<u>Foobar is foobar and void is </u>"
/>
`;
